File: server/PQL-Domain-Meta/src/main/java/com/passql/meta/entity/Legal.java
package com.passql.meta.entity;

import com.passql.common.entity.BaseEntity;
import com.passql.meta.constant.LegalStatus;
import com.passql.meta.constant.LegalType;
import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Entity
@Table(
    name = "legal",
    uniqueConstraints = {
        @UniqueConstraint(name = "uk_legal_type", columnNames = "type")
    }
)
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class Legal extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(updatable = false, nullable = false)
    private UUID legalUuid;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private LegalType type;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private LegalStatus status;
}

File: client/src/api/legal.ts
import { apiFetch } from "./client";

export type LegalType = "TERMS_OF_SERVICE" | "PRIVACY_POLICY";
export type LegalStatus = "PUBLISHED" | "DRAFT";

export interface LegalResponse {
  readonly type: LegalType;
  readonly title: string;
  readonly content: string;
  readonly status: LegalStatus;
}

export function fetchLegal(type: LegalType): Promise<LegalResponse> {
  return apiFetch<LegalResponse>(`/meta/legal/${type}`);
}
